app/controllers/decidim/vocdoni/elections_controller.rb

```diff
@@ -28,6 +37,32 @@ def election
         @election ||= Decidim::Vocdoni::Election.where(component: current_component).find(params[:id])
       end
 
+      def current_vocdoni_wallet
+        @current_vocdoni_wallet ||= Decidim::Vocdoni::Wallet.find_by(decidim_organization_id: current_organization.id)
+      end
+
+      def api_endpoint_env
+        @api_endpoint_env ||= Decidim::Vocdoni.api_endpoint_env
+      end
+
+      def vocdoni_client
+        @vocdoni_client ||= VocdoniClient.new(wallet: current_vocdoni_wallet.private_key, api_endpoint_env: api_endpoint_env)
+      end
+
+      def vocdoni_election_id
+        @vocdoni_election_id ||= election.vocdoni_election_id
+      end
+
+      def election_data
+        @election_data ||= vocdoni_client.fetch_election(vocdoni_election_id)
+      end
+
+      def election_metadata
+        return nil unless !election.election_type["secret_until_the_end"] && election.ongoing?
+
+        @election_metadata ||= election_data
+      end
+
```

Comment:
Related to my other comments, I think we can simplify this a lot: 

```suggestion
      def vocdoni_client
        @vocdoni_client ||= VocdoniClient.new(vocdoni_election_id: election.vocdoni_election_id)
      end

      def election_metadata
        return nil unless !election.election_type["secret_until_the_end"] && election.ongoing?

        @election_metadata ||= vocdoni_client.fetch_election
      end
```